Foundation/NSJSONSerialization.swift

```diff
@@ -271,19 +281,24 @@ private struct JSONWriter {
     
     mutating func serializeJSON(_ obj: Any) throws {
 
+        // For better performance, the most expensive conditions to evaluate should be last.
         switch (obj) {
         case let str as String:
             try serializeString(str)
         case let boolValue as Bool:
             serializeBool(boolValue)
-        case _ where _SwiftValue.store(obj) is NSNumber:
-            try serializeNumber(_SwiftValue.store(obj) as! NSNumber)
+        case let num as Int:
+            writer(String(describing: num))
```

Comment:
It's the `String(describing:)` output that I'm concerned about, yes. I would rather control the output in code more tightly coupled to the writer itself, since formatting that output exactly is basically the entire job of this class.